x-pack/plugins/apm/public/hooks/use_debounce.tsx

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { debounce } from 'lodash';
+import { useCallback, useState } from 'react';
+
+export function useStateDebounced<T>(initialValue: T, debounceDelay: number) {
+  const [debouncedValue, setValue] = useState(initialValue);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const setValueDebounced = useCallback(debounce(setValue, 300), [setValue]);
```

Comment:
Just noticed you're not using the `debounceDelay` argument here.

Reply:
Good catch!